File: src/main/java/seremis/soulcraft/inventory/slot/ToggleableMoveSlot.java
package seremis.soulcraft.inventory.slot;

import net.minecraft.inventory.IInventory;

public class ToggleableMoveSlot extends ToggleableSlot {

    private int srcX;
    private int srcY;

    public ToggleableMoveSlot(IInventory inventory, int id, int x, int y) {
        super(inventory, id, x, y);
        srcX = x;
        srcY = y;
    }

    @Override
    public void enable() {
        enabled = true;
        xDisplayPosition = srcX;
        yDisplayPosition = srcY;
    }

    @Override
    public void disable() {
        enabled = false;
        xDisplayPosition = srcX + 1000;
        yDisplayPosition = srcY + 1000;
    }
}

File: src/main/scala/seremis/geninfusion/inventory/slot/ToggleableMoveSlot.scala
package seremis.geninfusion.inventory.slot

import net.minecraft.inventory.IInventory

class ToggleableMoveSlot(inventory: IInventory, id: Int, x: Int, y: Int) extends ToggleableSlot(inventory, id, x, y) {

    override def enable() {
        enabled = true
        xDisplayPosition = x
        yDisplayPosition = y
    }

    override def disable() {
        enabled = false
        xDisplayPosition = x + 1000
        yDisplayPosition = y + 1000
    }
}